src/odemis/gui/comp/viewport.py

```diff
@@ -1065,7 +1066,334 @@ def _on_peak_method(self, state):
+        zoom_centre = self.canvas.pos_y_to_val_y(evt.Position[1])
+        scale = 0.1
+        rng = self.vrange.value
+
+        # proportion of zoom_centre
+        prop = (zoom_centre - rng[0]) / span
+
+        # zoom around the centre point
+        if rot < 0:
+            new_span = span * (1 + scale)
+        else:
+            new_span = span * (1 - scale)
+
+        lo = zoom_centre - prop * new_span
+        hi = lo + new_span
+
+        # Clamp the ranges - this functions differently for scrolls than pans
+        if lo < self.canvas.data_yrange[0]:
+            lo = self.canvas.data_yrange[0]
+        if hi > self.canvas.data_yrange[1] + self.canvas.data_yrange[1] * self._vmargin:
+            hi = self.canvas.data_yrange[1] + self.canvas.data_yrange[1] * self._vmargin
+
+        self.vrange.value = (lo, hi)
+
+    def on_drag_start(self, evt):
+        """ Start a dragging procedure """
+        if not self.canvas.has_data():
+            return
+
+        # Fixme: only go to drag mode if the mouse moves before a mouse up?
+        self._dragging = True
+
+        pos = evt.Position
+        self.drag_init_pos = (self.canvas.pos_x_to_val_x(pos[0]),
+                              self.canvas.pos_y_to_val_y(pos[1]))
+
+        logging.debug("Drag started at %s", self.drag_init_pos)
+
+    def on_drag_end(self, evt):
+        """ End the dragging procedure """
+        if not self.canvas.has_data():
+            return
+
+        # If the canvas was being dragged
+        if self._dragging:
+            self._dragging = False
+
+        self.was_dragged = False
+
+    def on_hlegend_motion(self, evt):
+        """ Process mouse motion
+
+        Set the drag shift and refresh the image if dragging is enabled and the left mouse button is
+        down.
+
+        """
+        if self._dragging:
+
+            v_pos = (self.canvas.pos_x_to_val_x(evt.Position[0]),
+                self.canvas.pos_y_to_val_y(evt.Position[1]))
+
+            self.drag_shift = (v_pos[0] - self.drag_init_pos[0],
+                          v_pos[1] - self.drag_init_pos[1])
+
+            # Rescale the horizontal axis
+            (lo, hi) = self.hrange.value
+
+            lo += (self.drag_shift[0] * self._hdrag_scale_factor)
+            hi += (self.drag_shift[0] * self._hdrag_scale_factor)
+
+            if lo < self.canvas.data_xrange[0]:
+                lo = self.canvas.data_xrange[0]
+                hi = self.canvas.display_xrange[1]
+            if hi > self.canvas.data_xrange[1]:
+                hi = self.canvas.data_xrange[1]
+                lo = self.canvas.display_xrange[0]
+
+            self.hrange.value = (lo, hi)
+
+    def on_vlegend_motion(self, evt):
+        """ Process mouse motion
+
+        Set the drag shift and refresh the image if dragging is enabled and the left mouse button is
+        down.
+
+        """
+        if self._dragging:
+
+            v_pos = (self.canvas.pos_x_to_val_x(evt.Position[0]),
```

Comment:
You don't really need `v_pos[0]`, do you? If so, you can skip its computation. Same thing for `on_hlegend_motion()`